### PR TITLE
Implemented clamp and smootherstep to MathUtils 

### LINKED ISSUE #13231: clamp and smootherstep in MathUtils
## Reason
There are often times where material properties transition to different regimes. A smooth transition between the regimes helps convergence.

## Design
```
Real smootherstep(Real start, Real end) 
{
  Real x = clamp((x - start) / (end - start), 0.0, 1.0);
  return Utility::pow<3>(x) * (x * (x * 6.0 - 15.0) + 10.0);
}

Real clamp(Real x, Real lowerlimit, Real upperlimit) 
{
  if (x < lowerlimit)
    return lowerlimit;
  if (x > upperlimit)
    return upperlimit;
  return x;
}
```

## Impact
This will provide a standard way of smoothing.

### PR DESCRIPTION
**Pull request by tophmatthews**
closes #13231

### CODE REVIEW

File: framework/include/utils/MathUtils.h [hunk 1]
```diff
@@ -131,6 +131,35 @@ poly(std::vector<Real> c, const T x, const bool derivative)
...
+  if (x > upperlimit)
+    return upperlimit;
+  return x;
+}
+
+template <typename T>
+T
+smootherStep(T x, T start, T end, bool derivative = false)
+{
+  x = clamp((x - start) / (end - start), 0.0, 1.0);
```
Comment (permcody) [inline framework/include/utils/MathUtils.h]: You have a few potential division by zeros in here. I would recommend a `mooseAssert()` or some other explicit parameter check.

File: test/include/functions/ClampTestFunction.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+template <>
+InputParameters validParams<ClampTestFunction>();
+
+class ClampTestFunction : public Function
+{
+public:
+  ClampTestFunction(const InputParameters & parameters);
+
+  virtual Real value(Real t, const Point & p);
```
Comment (permcody) [inline test/include/functions/ClampTestFunction.h]: override

File: test/include/functions/SmootherStepTestFunction.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+template <>
+InputParameters validParams<SmootherStepTestFunction>();
+
+class SmootherStepTestFunction : public Function
+{
+public:
+  SmootherStepTestFunction(const InputParameters & parameters);
+
+  virtual Real value(Real t, const Point & p);
```
Comment (permcody) [inline test/include/functions/SmootherStepTestFunction.h]: override

**Review by permcody — CHANGES_REQUESTED** [review]
A couple of minor issues.

File: framework/include/utils/MathUtils.h (hunk 1 above)
Comment (tophmatthews) [reply framework/include/utils/MathUtils.h]: Good point! I just returned 0.0 if start == end

**Review by permcody — APPROVED** [review]

**Pull request merged**